splunk-otel-android/src/main/java/com/splunk/rum/Carrier.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import android.os.Build;
+import android.telephony.TelephonyManager;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import java.util.Objects;
+
+@RequiresApi(api = Build.VERSION_CODES.P)
+class Carrier {
+
+    private final int id;
+    private final @Nullable String name;
+    private final @Nullable String mobileCountryCode; // 3 digits
+    private final @Nullable String mobileNetworkCode; // 2 or 3 digits
+    private final @Nullable String isoCountryCode;
+
+    static Builder builder() {
+        return new Builder();
+    }
+
+    Carrier(Builder builder) {
+        this.id = builder.id;
+        this.name = builder.name;
+        this.mobileCountryCode = builder.mobileCountryCode;
+        this.mobileNetworkCode = builder.mobileNetworkCode;
+        this.isoCountryCode = builder.isoCountryCode;
+    }
+
+    int getId() {
+        return id;
```

Comment:
Is `id` used anywhere in the code? Or is it just for `equals()`/`hashCode()` comparisons?

Reply:
It's not. I thought initially we might ingest it as well, but since we have the other components we could drop it. How strongly do you feel about getting rid of it? Seems fairly harmless.

Reply:
I don't feel that strongly about it; I just thought "is this one even used?"